storage/types.go

```diff
@@ -79,16 +80,39 @@ func NewNodeIDWithPrefix(prefix uint64, prefixLenBits, nodeIDLenBits, maxLenBits
 	return p
 }
 
+func bitLen(x int64) int {
+	r := 0
+	for x > 0 {
+		r++
+		x >>= 1
+	}
+	return r
+}
+
 // NewNodeIDForTreeCoords creates a new NodeID for a Tree node with a specified depth and
-// index
-func NewNodeIDForTreeCoords(depth int64, index int64, maxLenBits int) NodeID {
-	r := NewEmptyNodeID(maxLenBits)
-	for i := len(r.Path) - 1; index > 0; i-- {
+// index.
+// This method is used exclusively by the Log, and since the Log model grows upwards from the
+// leaves we modify the provided coords accordingly.
+//
+// index is the horizontal index into the tree at level depth, so the returned
+// NodeID will be zero padded on the right by depth places.
+func NewNodeIDForTreeCoords(depth int64, index int64, maxTreeDepth int) (NodeID, error) {
+	bl := bitLen(index)
+	if depth < 0 || bl > int(maxTreeDepth-int(depth)) {
```

Comment:
In that case, clarify what the parameter depth represents in the function comment?  Pretty please?